src/internal/direction/index.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { PointerEvent as ReactPointerEvent } from 'react';
+
+export function getIsRtl(element: null | HTMLElement | SVGElement): boolean {
+  if (!element) {
+    return false;
+  }
+  return getComputedStyle(element).direction === 'rtl';
+}
+
+export function getOffsetInlineStart(element: HTMLElement) {
+  const offsetParentWidth = element.offsetParent?.clientWidth ?? 0;
+  return getIsRtl(element) ? offsetParentWidth - element.offsetWidth - element.offsetLeft : element.offsetLeft;
+}
+
+/**
+ * The scrollLeft value will be a negative number if the direction is RTL and
+ * needs to be converted to a positive value for direction independent scroll
+ * computations. Additionally, the scrollLeft value can be a decimal value on
+ * systems using display scaling requiring the floor and ceiling calls.
+ */
```

Comment:
Why do we round the value? Isn't the decimal value more accurate, i.e. better for calculations?

---

> needs to be converted

let's document whether this function does that or the consumer of this function has to do it

```suggestion
/**
 * The scrollLeft value returned by the browser will be a negative number if the direction is RTL.
 * This function returns a positive value for direction independent scroll
 * computations. Additionally, the scrollLeft value can be a decimal value on
 * systems using display scaling requiring the floor and ceiling calls.
 */
```